libsolidity/ast/Types.cpp

```diff
@@ -2698,6 +2708,113 @@ Type const* TupleType::mobileType() const
 	return TypeProvider::tuple(move(mobiles));
 }
 
+BoolResult InlineArrayType::isImplicitlyConvertibleTo(Type const& _other) const
+{
+	auto arrayType = dynamic_cast<ArrayType const*>(&_other);
+
+	if (!arrayType || arrayType->isByteArrayOrString())
+		return false;
+	else
+	{
+		if (!arrayType->isDynamicallySized() && arrayType->length() != components().size())
+			return false; //TODO return string
+
+		for (Type const* c: components())
+			if (!c->isImplicitlyConvertibleTo(*arrayType->baseType()))
+				return false;
+
+		return true;
+	}
+}
+
+string InlineArrayType::richIdentifier() const
+{
+	return "t_inline_array" + identifierList(components());
+}
+
+bool InlineArrayType::operator==(Type const& _other) const
+{
+	if (auto inlineArrayType = dynamic_cast<InlineArrayType const*>(&_other))
+		// TODO: raise issue - do not compare by pointer for tuple type
+		return components() == inlineArrayType->components();
+	else
+		return false;
+}
+
+string InlineArrayType::toString(bool _short) const
+{
+	vector<string> result;
+
+	for (auto const& t: components())
+		result.push_back(t->toString(_short));
+
+	// TODO joinHumanReadable - is it fine to have a space here?
+	return "inline_array(" + util::joinHumanReadable(result) + ")";
+}
+
+u256 InlineArrayType::storageSize() const
+{
+	solAssert(false, "Storage size of non-storable InlineArrayType type requested.");
+}
+
+Type const* InlineArrayType::mobileType() const
+{
+	TypePointers mobiles;
+	for (auto const& c: components())
+	{
+		if (c)
+		{
+			auto mt = c->mobileType();
+			if (!mt)
+				return nullptr;
+			mobiles.push_back(mt);
+		}
+		else
+			mobiles.push_back(nullptr);
+	}
+	return TypeProvider::array(
+		DataLocation::Memory,
+		componentsCommonMobileType(),
+		components().size()
+	);
+}
+
+Type const* InlineArrayType::componentsCommonMobileType() const
+{
+	Type const* commonType = nullptr;
+	for (Type const* type: m_components)
+		commonType = commonType ?
+					Type::commonType(commonType, type->mobileType()) : type->mobileType();
+	return TypeProvider::withLocationIfReference(DataLocation::Memory, commonType);
+}
+
+Type const* InlineArrayType::fullEncodingType(bool _inLibraryCall, bool _encoderV2, bool _packed) const
+{
+	ArrayType const& arrayType = *TypeProvider::array(
+		DataLocation::Memory,
+		componentsCommonMobileType(),
+		components().size()
+	);
+
+	return arrayType.fullEncodingType(_inLibraryCall, _encoderV2, _packed);
```

Comment:
```suggestion
	return TypeProvider::array(
		DataLocation::Memory,
		componentsCommonMobileType(),
		components().size()
	)->fullEncodingType(_inLibraryCall, _encoderV2, _packed);
```

Reply:
Ah but wait - do we still need this override it all? The default implementation calls `mobileType` on the inline array, so I think the result would be the same.